<launch>
    <node pkg="pr2_pick_and_place_demos" name="pr2_pick_and_place_keyboard_interface" 
      type="pick_and_place_keyboard_interface" respawn="false" output="screen">

        <param name="use_database" value="true" />

        <remap from="right_arm/constraint_aware_ik" to="/pr2_right_arm_kinematics/get_constraint_aware_ik" />    
        <remap from="left_arm/constraint_aware_ik" to="/pr2_left_arm_kinematics/get_constraint_aware_ik" />    

        <remap from="right_arm/get_ik_solver_info" to="/pr2_right_arm_kinematics/get_ik_solver_info" />    
        <remap from="left_arm/get_ik_solver_info" to="/pr2_left_arm_kinematics/get_ik_solver_info" />    

        <remap from="right_arm/get_state_validity" to="/environment_server_right_arm/get_state_validity" />    
        <remap from="left_arm/get_state_validity" to="/environment_server_left_arm/get_state_validity" />    

        <remap from="right_arm/move_arm" to="/move_right_arm" />    
        <remap from="left_arm/move_arm" to="/move_left_arm" />    

        <remap from="right_arm/hand_posture_execution" to="/r_gripper_grasp_posture_controller" />    
        <remap from="left_arm/hand_posture_execution" to="/l_gripper_grasp_posture_controller" />    

	<remap from="right_arm/grasp_status" to="/r_gripper_grasp_status" />
      	<remap from="left_arm/grasp_status" to="/l_gripper_grasp_status" />

        <remap from="right_arm/cart/command_pose" to="/r_cart/command_pose" />
        <remap from="left_arm/cart/command_pose" to="/l_cart/command_pose" />

        <remap from="switch_controller" to="pr2_controller_manager/switch_controller"/>
        <remap from="list_controllers" to="pr2_controller_manager/list_controllers"/>

	<param name="right_cartesian_controller" value="r_cart" /> 
        <param name="left_cartesian_controller" value="l_cart" />
        <param name="right_joint_controller" value="r_arm_controller" />
        <param name="left_joint_controller" value="l_arm_controller" />
    </node>
</launch>

